### DISCUSSION TITLE
[javascript deep dive] 12장. 함수

### DISCUSSION THREAD

**Discussion by rladpwl0512** [woowacourse-study/2022-javascript-bank#17] [General]
[javascript deep dive] 12장. 함수

**Comment by jswith** [comment]
함수표현식과 함수선언문의 차이는 무엇인가요?

**Comment by jswith** [comment]
화살표 함수와 일반 함수의 차이는 무엇인가요?

**Comment by InKyoJeong** [comment]
자바스크립트 함수는 일급 객체라고 하는데, 일급객체란 무엇일까? 함수가 일급 객체이기 때문에 할 수 있는것은 무엇일까?

**Comment by rladpwl0512** [comment]
함수 선언문을 통해 생성한 함수를 호출할 수 있는 이유를 자바스크립트 엔진 내부동작과 연결하여 설명해주세요. 
(함수 선언문을 생성할 때는 식별자를 따로 선언, 할당하지 않는데 어떻게 호출이 가능한가요?)

**Comment by rladpwl0512** [comment]
함수선언문, 함수 표현식의 호이스팅을 비교해서 설명해주세요.

**Comment by rladpwl0512** [comment]
Javascript 함수의 반환문(return문)의 역할 두가지를 설명해주세요.

**Comment by rladpwl0512** [comment]
재귀함수의 개념과, 재귀함수 사용의 장단점을 설명해주세요.

**Comment by rladpwl0512** [comment]
고차함수, 콜백함수란 무엇인가요?

**Comment by rladpwl0512** [comment]
순수함수, 비순수함수란 무엇인가요?

**Comment by rladpwl0512** [comment]
[todo 공부...] 
- 함수형 프로그래밍, 객체지향 프로그래밍, 절차지향 프로그래밍에 대해 설명해주세요. 
- Javascript는 어디에 속하나요?
